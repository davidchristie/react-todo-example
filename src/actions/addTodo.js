export const ADD_TODO = 'ADD_TODO'

export default newTodo => ({
  payload: newTodo,
  type: ADD_TODO
})
