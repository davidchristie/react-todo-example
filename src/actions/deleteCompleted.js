export const DELETE_COMPLETED = 'DELETE_COMPLETED'

export default () => ({
  type: DELETE_COMPLETED
})
