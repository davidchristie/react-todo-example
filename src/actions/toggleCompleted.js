export const TOGGLE_COMPLETED = 'TOGGLE_COMPLETED'

export default itemIndex => ({
  payload: itemIndex,
  type: TOGGLE_COMPLETED
})
