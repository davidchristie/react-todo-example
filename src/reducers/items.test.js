import addTodo from '../actions/addTodo'
import deleteCompleted from '../actions/deleteCompleted'
import toggleCompleted from '../actions/toggleCompleted'
import reducer from './items'

it('adds a new todo item to state', () => {
  const state = [
    {
      completed: false,
      text: 'Test item 1'
    },
    {
      completed: false,
      text: 'Test item 2'
    }
  ]
  const action = addTodo({
    completed: false,
    text: 'Test item 3'
  })
  expect(reducer(state, action)).toMatchSnapshot()
})

it('removes completed todo item from state', () => {
  const state = [
    {
      completed: true,
      text: 'Test item 1'
    },
    {
      completed: false,
      text: 'Test item 2'
    },
    {
      completed: true,
      text: 'Test item 3'
    }
  ]
  const action = deleteCompleted()
  expect(reducer(state, action)).toMatchSnapshot()
})

it('toggles todo item completion state', () => {
  const state = [
    {
      completed: false,
      text: 'Test item 1'
    },
    {
      completed: false,
      text: 'Test item 2'
    },
    {
      completed: false,
      text: 'Test item 3'
    }
  ]
  const action = toggleCompleted(1)
  expect(reducer(state, action)).toMatchSnapshot()
})
