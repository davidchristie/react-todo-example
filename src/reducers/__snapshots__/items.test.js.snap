// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`adds a new todo item to state 1`] = `
Array [
  Object {
    "completed": false,
    "text": "Test item 1",
  },
  Object {
    "completed": false,
    "text": "Test item 2",
  },
  Object {
    "completed": false,
    "text": "Test item 3",
  },
]
`;

exports[`removes completed todo item from state 1`] = `
Array [
  Object {
    "completed": false,
    "text": "Test item 2",
  },
]
`;

exports[`toggles todo item completion state 1`] = `
Array [
  Object {
    "completed": false,
    "text": "Test item 1",
  },
  Object {
    "completed": true,
    "text": "Test item 2",
  },
  Object {
    "completed": false,
    "text": "Test item 3",
  },
]
`;
